app/javascript/components/ingredients/base-component.vue

```diff
@@ -77,39 +86,107 @@
 
 <script>
 
+import { getIngredients, postIngredient, deleteIngredient, editIngredient } from './../../api/ingredients.js';
+
 export default {
-  props: {
-    ingredients: { type: Array, required: true },
-  },
+
   data() {
     return {
       showingAdd: false,
       showingEdit: false,
       showingDel: false,
       ingredientToEdit: {},
+      ingredientToDelete: {},
       tableHeader: ['name', 'price', 'quantity', 'measure'],
+      ingredients: [],
+      status: '',
+      error: '',
     };
   },
 
+  async created() {
+    try {
+      const response = await getIngredients();
+      this.ingredients = response.data.data.map((element) => ({
+        id: element.id,
+        ...element.attributes,
+      }));
+      this.successResponse(response);
+    } catch (error) {
+      this.errorResponse(error);
+    }
+  },
+
   methods: {
     toggleAddModal() {
       this.showingAdd = !this.showingAdd;
     },
+
     toggleEditModal(ingredient) {
       this.showingEdit = !this.showingEdit;
       this.ingredientToEdit = ingredient;
     },
-    toggleDelModal() {
+
+    toggleDelModal(ingredient) {
       this.showingDel = !this.showingDel;
+      this.ingredientToDelete = ingredient;
     },
-    addIngredient() {
+
+    async addIngredient() {
       this.showingAdd = !this.showingAdd;
+
+      try {
+        const {
+          status,
+          data:
+            { data: { id, attributes },
+            },
+        } = await postIngredient(this.$refs.addIngredientInfo.form);
+        const ingredientToAdd = { id, ...attributes };
+        this.ingredients.push(ingredientToAdd);
+        this.successResponse(status);
+      } catch (error) {
+        this.errorResponse(error);
+      }
     },
-    editIngredient() {
+
+    async editIngredient() {
       this.showingEdit = !this.showingEdit;
+      try {
+        const res = await editIngredient(this.ingredientToEdit.id, this.$refs.editIngredientInfo.form);
```

Comment:
Que es lo que se está haciendo acá? No entiendo este `this.$refs.editIngredientInfo.form`